Microsoft/WebTemplateStudio
# Create resource group for Azure Functions

Related PBI: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/29740

The purpose of this PR is to create resource group for user if they choose to put the Azure Functions they create in a new resource group. 

Client side changes will be in another PR (the client will be sending an empty string for `functions.resourceGroup` field in the payload when user choose a new resource). 

So, to test for this PR:
- Add this line `payload.functions.resourceGroup = "";` before line 92 in `generationExperience.ts`. This will force the function that deploys creation of resource group to be kicked off. 
- Launch the wizard
- Take note of the project name you entered in the first page of the wizard. 
- Add Azure Functions when creating your project (the resource group you choose in the form does not matter)
- Go onto Azure portal and check if a new resource group has been created. The name of the new resource group should start with the project name your entered. The Azure functions you created should be in that resource group.

**Note**: Logics that handles creation of resource group for Cosmos (which may be the same as the resource group for Functions) will be handled in another PR. The arm template required for resource group in the generated project will also be addressed in another PR.
